core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -672,6 +672,9 @@ public int getNumberOfCores() {
           .getProperty(CarbonCommonConstants.NUM_CORES_LOADING,
               CarbonCommonConstants.NUM_CORES_DEFAULT_VAL));
```

Comment:
Since now we always update the CarbonCommonConstants.NUM_CORES_LOADING value in LoadCommand, we should not use default value here. You can use `String getProperty(String key)` directly.